routemaster/models/event.rb

```diff
@@ -12,24 +13,26 @@ class Event < Message
 
       VALID_TYPES = %w(create update delete noop)
 
-      attr_reader :topic, :type, :url
+      attr_reader :topic, :type, :url, :data
 
       def initialize(**options)
         super
-        _assert VALID_TYPES.include?(options[:type]), 'bad event type'
+        @type      = options.fetch(:type, nil)
+        @url       = CallbackURL.new options.fetch(:url, nil)
```

Comment:
It's not allowed to be `nil` (as the spec shows!)
The code is a little misleading — it's the value object wrapper `CallbackURL` that'll handle nils (and bark) in its constructor.